src/Plugin.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+/**
+ * @file
+ * Contains \Netzstrategen\Varnish\Plugin.
+ */
+
+namespace Netzstrategen\Varnish;
+
+/**
+ * Main front-end functionality.
+ */
+class Plugin {
+
+  /**
+   * Prefix for naming.
+   *
+   * @var string
+   */
+  const PREFIX = 'varnish';
+
+  /**
+   * Gettext localization domain.
+   *
+   * @var string
+   */
+  const L10N = self::PREFIX;
+
+  /**
+   * @var string
+   */
+  private static $baseUrl;
+
+  /**
+   * Loads the plugin textdomain.
+   */
+  public static function loadTextdomain() {
+    load_plugin_textdomain(static::L10N, FALSE, static::L10N . '/languages/');
+  }
+
+  /**
+   * @implements init
+   */
+  public static function init() {
+    add_action('save_post', __CLASS__ . '::purgePost', 10, 2);
+    add_action('deleted_post', __CLASS__ . '::purgePost', 10, 2);
+    add_action('trashed_post', __CLASS__ . '::purgePost', 10, 2);
+    add_action('edit_post', __CLASS__ . '::purgePost', 10, 2);
+    add_action('delete_attachment', __CLASS__ . '::purgePost', 10, 2);
+
+    add_action('switch_theme', __CLASS__ . '::purgePost', 10, 2);
+
+    add_action('wp_update_attachment_metadata', __CLASS__ . '::purgeAttachmentMeta', 50, 2);
+
+    add_action('shutdown', __CLASS__ . '::executePurge');
+  }
+
+  /**
+   * Registers the given post, associated terms, authors, and feeds for purging.
+   *
+   * @param int $postId
+   *   The ID of the post to purge.
+   */
+  public static function purgePost($postId) {
+    // Skip post revisions.
+    if (get_permalink($postId) === FALSE) {
+      return;
+    }
+    // Post URL.
+    Varnish::$purgeUrls[get_permalink($postId)] = FALSE;
+    // Post pagination, feeds, and other attached URLs.
+    Varnish::$purgeUrls[get_permalink($postId) . '/.*'] = TRUE;
+
+    // All associated public terms (categories, tags, custom taxonomies).
+    foreach (wp_get_object_terms($postId, get_taxonomies(['public' => TRUE])) as $term) {
+      $term_link = get_term_link($term);
+      Varnish::$purgeUrls[$term_link] = FALSE;
+      Varnish::$purgeUrls[$term_link . '/page/.*'] = TRUE;
+      Varnish::$purgeUrls[$term_link . '/feed'] = FALSE;
+
+      // Walk up parent terms in the hierarchy.
+      while ($term->parent) {
+        $term = get_term($term->parent, $term->taxonomy);
+        $term_link = get_term_link($term);
+        Varnish::$purgeUrls[$term_link] = FALSE;
+        Varnish::$purgeUrls[$term_link . '/page/.*'] = TRUE;
+        Varnish::$purgeUrls[$term_link . '/feed'] = FALSE;
+      }
+    }
+
+    // Author URL.
+    $author_id = get_post_field('post_author', $postId);
+    Varnish::$purgeUrls[$author_url = get_author_posts_url($author_id)] = FALSE;
```

Comment:
While this is a handy writing, I'd suggest to move the `$author_url` to a separate line to keep consistency and maintain easy readability (similar to https://github.com/netzstrategen/wordpress-varnish/pull/1/files#diff-f7fcf01b75ab0236c970de527853c3abR76)